pkg/plugins/policies/meshcircuitbreaker/api/v1alpha1/meshcircuitbreaker.go

```diff
@@ -0,0 +1,157 @@
+type To struct {
+	// TargetRef is a reference to the resource that represents a group of
+	// destinations.
+	TargetRef common_api.TargetRef `json:"targetRef,omitempty"`
+	// Default is a configuration specific to the group of destinations
+	// referenced in 'targetRef'
+	Default Conf `json:"default,omitempty"`
+}
+
+type From struct {
+	// TargetRef is a reference to the resource that represents a group of
+	// destinations.
+	TargetRef common_api.TargetRef `json:"targetRef,omitempty"`
+	// Default is a configuration specific to the group of destinations
+	// referenced in 'targetRef'
+	Default Conf `json:"default,omitempty"`
+}
+
+type Conf struct {
+	ConnectionLimits *ConnectionLimits `json:"connectionLimits,omitempty"`
+	OutlierDetection *OutlierDetection `json:"outlierDetection,omitempty"`
+}
+
+type ConnectionLimits struct {
+	// The maximum number of connections allowed to be made to the upstream
+	// cluster.
+	MaxConnections *uint32 `json:"maxConnections,omitempty"`
+	// The maximum number of connection pools per cluster that are concurrently
+	// supported at once. Set this for clusters which create a large number of
+	// connection pools.
+	MaxConnectionPools *uint32 `json:"maxConnectionPools,omitempty"`
+	// The maximum number of pending requests that are allowed to the upstream
+	// cluster. This limit is applied as a connection limit for non-HTTP
+	// traffic.
+	MaxPendingRequests *uint32 `json:"maxPendingRequests,omitempty"`
+	// The maximum number of parallel retries that will be allowed to
+	// the upstream cluster.
+	MaxRetries *uint32 `json:"maxRetries,omitempty"`
+	// The maximum number of parallel requests that are allowed to be made
+	// to the upstream cluster. This limit does not apply to non-HTTP traffic.
+	MaxRequests *uint32 `json:"maxRequests,omitempty"`
+}
+
+type OutlierDetection struct {
+	// When set to true, outlierDetection configuration won't take any effect
+	Disabled *bool `json:"disabled,omitempty"`
+	// The time interval between ejection analysis sweeps. This can result in
+	// both new ejections and hosts being returned to service.
+	Interval *k8s.Duration `json:"interval,omitempty"`
+	// The base time that a host is ejected for. The real time is equal to
+	// the base time multiplied by the number of times the host has been
+	// ejected.
+	BaseEjectionTime *k8s.Duration `json:"baseEjectionTime,omitempty"`
+	// The maximum % of an upstream cluster that can be ejected due to outlier
+	// detection. Defaults to 10% but will eject at least one host regardless of
+	// the value.
+	MaxEjectionPercent *uint32 `json:"maxEjectionPercent,omitempty"`
+	// Determines whether to distinguish local origin failures from external
+	// errors. If set to true the following configuration parameters are taken
+	// into account: detectors.localOriginFailures.consecutive
+	SplitExternalAndLocalErrors *bool      `json:"splitExternalAndLocalErrors,omitempty"`
+	Detectors                   *Detectors `json:"detectors,omitempty"`
```

Comment:
Docs? + Detectors struct

I know this is annoying, but we will generate nice docs / swagger form this.

Reply:
ack